src/ipc/Request.h

```diff
@@ -24,19 +25,20 @@ class Request: public RefCountable
     typedef RefCount<Request> Pointer;
 
 public:
-    Request(int aRequestorId, unsigned int aRequestId):
-        requestorId(aRequestorId), requestId(aRequestId) {}
+    Request(int aRequestorId, unsigned int aRequestId, bool initQuid):
+        requestorId(aRequestorId), requestId(aRequestId), qid(initQuid) {}
+
+    // no assignment of any kind, use clone() instead
+    Request &operator=(const Request &) = delete;
+    // TODO: also forbid move assignment
```

Comment:
It is not clear to me why

1. The comment implies that move assignment is not already forbidden when the compiler, AFAICT, cannot move-assign any Request object with the current code.
2. We do not implement this trivial TODO, removing the apparent contradiction with the "any kind" claim of the "no assignment" comment above.
3. We do not declare defaulted copy constructor and move constructors to be explicit about our needs/permissions instead of forcing folks to remember very complex member generation tables.

It feels like we are tiptoeing around something here, but I do not understand why (or what that "something" is). I know that we do not yet know why all those previously banned operations were previously banned, but we have to step into that unknown because, according to your earlier [comment](https://github.com/measurement-factory/squid/pull/110#discussion_r567116377), we need a previously banned copy constructor.

Reply:
I added explicit declarations for the copy constructor and deleted both assignment operators (31e2b13). Fortunately, it compiled without adjusting sublclasses.